client/onion_client.py

```diff
@@ -0,0 +1,145 @@
+import argparse             # for command-line argument parsing
+def shut_down_directory_node():
+    LISTEN_SOCKET.close()
+    if args.verbose: print('Client Node DOWN')
+
+import random               # for random path selection
+import socket               # for TCP communication
+import json                 # for encoding data sent through TCP
+import threading            # for one thread per TCP connection
+
+
+#   CLI ARGS
+###############################################################################
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("destination", help="hostname to connect to")
+parser.add_argument("port", help="port to listen on", type=int)
+parser.add_argument("-v", "--verbose",
+                    help="level of logging verbose", action="store_true")
+
+args = parser.parse_args()
+
+# validate args against conditions
+if args.port < 5551 or args.port > 5557: # 7 group members, each get a port
+    parser.error("port must satisfy: 5551 <= port <= 5557")
+if args.verbose: print('Verbosity ON\n')
```

Comment:
I don't think it's conventional to tell the user via `print` statement that they have in fact passed the `-v` flag.

Reply:
True indeed, fixed in 267e7f2